products/compute/api.yaml

```diff
@@ -11452,32 +11527,742 @@ objects:
+                              connecting to backend services, for example due to connection timeouts.  -
+                              retriable-4xx: Loadbalancer will retry for retriable 4xx response codes.
+                              Currently the only retriable error supported is 409.  - refused-
+                              stream:Loadbalancer will retry if the backend service resets the stream with a
+                              REFUSED_STREAM error code. This reset type indicates that it is safe to retry.
+                              - cancelledLoadbalancer will retry if the gRPC status code in the response
+                              header is set to cancelled  - deadline-exceeded: Loadbalancer will retry if the
+                              gRPC status code in the response header is set to deadline-exceeded  - resource-
+                              exhausted: Loadbalancer will retry if the gRPC status code in the response
+                              header is set to resource-exhausted  - unavailable: Loadbalancer will retry if
+                              the gRPC status code in the response header is set to unavailable
+                      - !ruby/object:Api::Type::NestedObject
+                        name: 'timeout'
+                        description: |
+                          Specifies the timeout for the selected route. Timeout is computed from the time
+                          the request is has been fully processed (i.e. end-of-stream) up until the
+                          response has been completely processed. Timeout includes all retries. If not
+                          specified, the default value is 15 seconds.
+                        properties:
+                          - !ruby/object:Api::Type::Integer
+                            name: 'nanos'
+                            description: |
+                              Span of time that's a fraction of a second at nanosecond resolution. Durations
+                              less than one second are represented with a 0 `seconds` field and a positive
+                              `nanos` field. Must be from 0 to 999,999,999 inclusive.
+                          - !ruby/object:Api::Type::String
+                            name: 'seconds'
+                            required: true
+                            description: |
+                              Span of time at a resolution of a second. Must be from 0 to 315,576,000,000
+                              inclusive.
+                      - !ruby/object:Api::Type::NestedObject
+                        name: 'urlRewrite'
+                        description: |
+                          The spec to modify the URL of the request, prior to forwarding the request to
+                          the matched service
+                        properties:
+                          - !ruby/object:Api::Type::String
+                            name: 'hostRewrite'
```

Comment:
`The value must be between 1 and 255 characters.` suggests the field is `Required`?

Reply:
I don't think so. There are different portions of the URL that can be replaced (`host`, `path`, and `prefix`) and none are required.